inst/js/gtIO.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Update the selected selectID when a row is clicked.
+ *
+ * @param {HTMLElement} row   - The table row that was clicked.
+ * @param {HTMLElement} table - The table element containing the row.
+ */
+const gtRowClickCallback = function(row, table) {
+  var selectID = extractSelectID(row);
+  if (selectID) {
+    var currentSelectID = $(table).data('selectID');
+    if (currentSelectID === selectID) {
+      selectID = "None";
+    }
+    $(table).data('selectID', selectID);
+    table.setAttribute('data-selectid', selectID);
+    $(table).trigger('gtIO-value-changed');
+  }
+};
+
+/**
+ * Bind click events to table rows and set up observers for selectID changes and
+ * table state changes.
+ *
+ * @param {HTMLElement} table - The table element containing the rows.
+ * @returns {boolean} True if rows were found and bound, false otherwise.
+ */
+const gtBindRowClickEvents = function(table) {
+  var rows = table.querySelectorAll(".rt-tbody .rt-tr");
+  if (rows.length > 0) {
+    rows.forEach(function(row) {
+      row.addEventListener('click', function() {
+        gtRowClickCallback(row, table);
+      });
+    });
+
+    const observer = new MutationObserver(function() {
+      gtUpdateRowHighlight(table);
+    });
+    observer.observe(table, { attributes: true, attributeFilter: ['data-selectid'] });
+
+    reactable_table = table.querySelector(".reactable");
+    window.Reactable.onStateChange(reactable_table.id, function(state) {
```

Comment:
if a site that is not currently visible in the reactable table due to pagination is selected in the sidebar, it would be nice if reactable would jump to that page to highlight the row. this can be a new issue, but if it's simple maybe it can just be handled here.